src/encode/util.rs

```diff
@@ -1,37 +1,34 @@
-use std::hash;
 use std::io;
 
 // A Write computing a digest on the bytes written.
-pub struct HasherWrite<'a, W, H>
+pub struct DigestWrite<'a, 'b, W, S>
```

Comment:
Extra `'b` lifetime is specifies that the lifetime of the mutable borrows (`'a`) is different from the lifetime of the CRC algorithm (`'b`) which `Digest` borrows immutably.